### PR TITLE
Rewritten Rejectionreport (routine report) in new UI !

### PR DESCRIPTION
**Pull request by manishjha-04**
**_Solving Issue_** : #757

**Before:**

Screenshot 2024-03-08 203243

**After:**

https://github.com/I-TECH-UW/OpenELIS-Global-2/assets/128405481/b10b418f-d1da-4353-a598-f3d73afe90fc

### CODE REVIEW

**Comment by manishjha-04** [comment]
Hey @mozzy11 please review !

 Also I wanted to ask should i create a generic component **_reportByDate_** for this? since it's being used multiple times within the Routine Report.  What do you think?

File: frontend/src/components/Reports/routine/RejectionReport.js
Comment (mherman22) [inline frontend/src/components/Reports/routine/RejectionReport.js:1]: Thanks @manishjha-04 for the work here!

I would add breadcrumbs here to make navigation back to the RoutineReports page less painful. something like `home / Routine Reports / Rejection Report`. You can pick a leaf at https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/783/commits/3a47438b17eadcc7cacea0f625d110c79ff8053d if needed!

**Review by mherman22 — CHANGES_REQUESTED** [review]

File: frontend/src/components/Reports/routine/RejectionReport.js
Comment (manishjha-04) [reply frontend/src/components/Reports/routine/RejectionReport.js:1]: Thanks a bunch, @mherman22! 🌟 It's a good idea ! I'm definitely going to implement it. 🚀

File: frontend/src/components/Reports/routine/RejectionReport.js [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+  const handleSubmit = () => {
+    setLoading(true);
+
+    const baseParams = 'report=sampleRejectionReport&type=indicator';
+
+    const baseUrl = `${config.serverBaseUrl}/ReportPrint`;
+    const url = `${baseUrl}?${baseParams}&upperDateRange=${reportFormValues.startDate}&lowerDateRange=${reportFormValues.endDate}`;
+
+    window.open(url, '_blank');
+
```
Comment (sherrif10) [inline frontend/src/components/Reports/routine/RejectionReport.js:67]: This looks good. One small thing, Remove these trailing spaces.

File: frontend/src/components/Reports/routine/RejectionReport.js [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    setReportFormValues(obj);
+  };
+
+  const handleSubmit = () => {
+    setLoading(true);
+
+    const baseParams = 'report=sampleRejectionReport&type=indicator';
+
+    const baseUrl = `${config.serverBaseUrl}/ReportPrint`;
+    const url = `${baseUrl}?${baseParams}&upperDateRange=${reportFormValues.startDate}&lowerDateRange=${reportFormValues.endDate}`;
```
Comment (konavivekramakrishna) [inline frontend/src/components/Reports/routine/RejectionReport.js:64]:
```suggestion
    const url = `${baseUrl}?${baseParams}&upperDateRange=${reportFormValues.endDate}&lowerDateRange=${reportFormValues.startDate}`;
```

Hello @manishjha-04 

IG the backend treats `upperDateRange` as `endDate` and `lowerDateRange` as `startDate`

just a minor suggestion

cc: @mozzy11

**Comment by konavivekramakrishna** [comment]
@manishjha-04 

there's a PR already merged https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/740/

with incorrect url `{baseParams}&upperDateRange=${reportFormValues.startDate}&lowerDateRange=${reportFormValues.endDate}`;`

ideally it should be as `{baseParams}&upperDateRange=${reportFormValues.endDate}&lowerDateRange=${reportFormValues.startDate}`;`

**Comment by manishjha-04** [comment]
> @manishjha-04
> 
> there's a PR already merged #740
> 
> with incorrect url `{baseParams}&upperDateRange=${reportFormValues.startDate}&lowerDateRange=${reportFormValues.endDate}`;`
> 
> ideally it should be as `{baseParams}&upperDateRange=${reportFormValues.endDate}&lowerDateRange=${reportFormValues.startDate}`;`

Ah yes! You are right @konavivekramakrishna I didnt notice only will make the changes and raise the PR ! Thanks man!

**Comment by konavivekramakrishna** [comment]
Thanks you too @manishjha-0 😄

actually I was working on this https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/766

so referred your previous pr,

and also can you please review my pr and let me know Ur suggestions as we do know have a new `reportByDate` component

**Comment by manishjha-04** [comment]
Indeed, you made a valid point in the  the PR #821 reportByDate component was introduced.
 I believe it would be beneficial to integrate this component consistently across all the reports. What are your thoughts on this, @mozzy11 @konavivekramakrishna !

**Comment by konavivekramakrishna** [comment]
Yup, @manishjha-04 

I will change the all the instances to use `reportBydate` component in https://github.com/I-TECH-UW/OpenELIS-Global-2/pull/766

File: frontend/src/components/Reports/routine/RejectionReport.js (hunk 2 above)
Comment (mozzy11) [reply frontend/src/components/Reports/routine/RejectionReport.js:64]: Thanks for catching this @konavivekramakrishna .

**Comment by mozzy11** [comment]
Thanks @manishjha-04

**Pull request merged**